src/binary_kind.ml

```diff
@@ -1,19 +1,22 @@
 open! Stdune
 
 type t =
+  | C
   | Exe
   | Object
   | Shared_object
 
 let dparse =
   let open Dsexp.Of_sexp in
   enum
-    [ "exe"           , Exe
+    [ "c"             , C
```

Comment:
@diml, technically, we should guard this behind the 1.2 version of the syntax. do you think it's worth it to complicate the parsing for this?

Reply:
Ah, that's a good point. I think it is worth complicating the parsing as it will help users of this feature. We should add support for this in `enum`, but in the meantime we can use `sum`. @drjdn could you replace `enum [...]` by:

```ocaml
sum
  [  "c", Syntax.since Stanza.syntax (1, 2) >>> return C
  ; "exe", return Exe
  ...
  ]
```

? This will ensure that users must write `(lang dune 1.2)` in their dune-project file to have access to `c` binary kind.

Reply:
I can easily do this but I'm not sure I follow the logic.  This works perfectly fine with dune 1.0.  It was just a feature that was added/turned on.  If it only worked correctly with 1.2 then sure but that is not the case.  If the next version of dune is 1.2 then this only lives on master until then. Anyone using a 1.* version would not have this feature anyway (I'm pretty sure almost all dune usage is using the opam package).

Reply:
Well we make it a conscious decision to opt in to use newer features of the dune language. I.e. we want to make sure that if a user writes `(lang dune 1.0)` in their dune project, he can safely assume that it will be build with dune `1.0.0`. This would clearly break this assumption.